Source/SwiftLintFramework/Rules/JoinedDefaultRule.swift

```diff
@@ -54,11 +58,66 @@ public struct JoinedDefaultParameterRule: ASTRule, ConfigurationProviderRule, Op
             let argumentBodyOffset = argument.bodyOffset,
             let argumentBodyLength = argument.bodyLength,
             argument.name == "separator"
-            else {
-                return nil
-        }
+            else { return nil }
 
         let body = file.contents.bridge().substringWithByteRange(start: argumentBodyOffset, length: argumentBodyLength)
         return body == "\"\"" ? argumentBodyOffset : nil
     }
+
+    // MARK: - CorrectableRule
+
+    public func correct(file: File) -> [Correction] {
+        let violatingRanges = violationRanges(in: file, dictionary: file.structure.dictionary)
+        let matches = file.ruleEnabled(violatingRanges: violatingRanges, for: self)
+        var correctedContents = file.contents
+        var adjustedLocations: [Int] = []
+
+        for violatingRange in matches.reversed() {
+            if let range = file.contents.nsrangeToIndexRange(violatingRange) {
+                correctedContents = correctedContents.replacingCharacters(in: range, with: "")
+                adjustedLocations.insert(violatingRange.location, at: 0)
+            }
+        }
+
+        file.write(correctedContents)
+
+        return adjustedLocations.map {
+            Correction(ruleDescription: type(of: self).description, location: Location(file: file, characterOffset: $0))
+        }
+    }
+
+    private func violationRanges(in file: File, dictionary: [String: SourceKitRepresentable]) -> [NSRange] {
+        return dictionary.substructure.flatMap { subDictionary -> [NSRange] in
+            let violations = violationRanges(in: file, dictionary: subDictionary)
+
+            guard
+                // is it calling a method '.joined' and passing a single argument?
+                subDictionary.kind == SwiftExpressionKind.call.rawValue,
+                subDictionary.name?.hasSuffix(".joined") == true,
+                subDictionary.enclosedArguments.count == 1
+                else { return violations }
+
+            guard
+                // is this single argument called 'separator'?
+                let argument = subDictionary.enclosedArguments.first,
+                let offset = argument.offset,
+                let length = argument.length,
+                argument.name == "separator"
+                else { return violations }
+
+            guard
+                // is this single argument the default parameter?
+                let bodyOffset = argument.bodyOffset,
+                let bodyLength = argument.bodyLength,
+                let body = file.contents.bridge().substringWithByteRange(start: bodyOffset, length: bodyLength),
```

Comment:
I've put these guards that have more computational cost at the end - `file.contents.bridge()...`.